File: backend/lib/domain/customer.rb
module Domain
  class Customer
    attr_reader :name, :email, :address, :memberships, :vouchers

    def initialize(attributes = {})
      @name, @email, @address = attributes.values_at(:name, :email, :address)
      @memberships = []
      @vouchers = []
    end

    def add_membership(membership)
      @memberships << membership
    end

    def add_voucher(voucher)
      @vouchers << voucher
    end
  end
end

File: backend/challenge-resolution/src/main/kotlin/core/domain/customer/Customer.kt
package main.kotlin.core.domain.customer

import main.kotlin.core.domain.voucher.Voucher

class Customer(val name: String){
    private val vouchers = mutableListOf<Voucher>()

    fun addVoucher(voucher: Voucher) {
        vouchers.add(voucher)
    }

    fun getVouchers() : List<Voucher>{
        return vouchers
    }
}